### PR TITLE
Added scraper for letterboxd, completes #1028 and fixes #1034

### LINKED ISSUE #1034: BUG: Typo in proxies attribute assignment in RequestConfig class
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

minor typo error in the proxies attribute assignment in `RequestConfig` class

image

### Record

- [X] I agree to follow this project's Code of Conduct
- [X] I'm a GSSoC'24 contributor
- [X] I want to work on this issue

### PR DESCRIPTION
**Pull request by arpy8**
## Description
1. Added the scraper for letterboxd website, supporting the following funtions:
	| Methods                     | Details                                                                                              |
    | --------------------------- | ---------------------------------------------------------------------------------------------------- |
    | `.films_watched()`          | Returns the numbers of films watched by the user.                                                    |
    | `.recent_activity(n)`       | Returns a list of length `n` of the latest activity by the user.                                     |
    | `.recent_reviews(n)`        | Returns a list of dictionaries of length `n` with the latest reviews by the user.                    |
    | `.get_watchlist(n)`         | Returns a list of length `n` including movies and series watchlisted by the user.                    |
    | `.get_followers_count()`    | Returns the number of followers of the user.                                                         |  
    | `.get_following_count()`    | Returns the number of following of the user.                                                         |

2. Fixed a minor typo error in the request_config.py file.

## Checklist

- [x] Have you adhered to the repository's defined coding convention rules?

## Screenshots
an example of the letterboxd scraper:
image

typo screenshot:
image

---

I certify that I have carried out the relevant checks and provided the requisite screenshot for validation by submitting this pull request.

### CODE REVIEW

File: src/scrape_up/letterboxd/letterboxd.py [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+        
+        Example:
+        ```
+        letterboxd_user = Letterboxd(username="arpy8")
+        letterboxd_user.recent_activity(n=5)
+        ```
+        
+        Returns a list of the latest activity by the user of length `n`.:
+        ```python
+        ['Challengers', 'Dune: Part Two', 'The Hunger Games: The Ballad of Songbirds & Snakes', 'BLUE EYE SAMURAI', 'Jaane Jaan']
```
Comment (nikhil25803) [inline src/scrape_up/letterboxd/letterboxd.py:64]: Instead of list, return in form of dictionary.

File: src/scrape_up/letterboxd/letterboxd.py [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+        try:
+            req = get(f"{self.url}/films/", self.config)
+
+            page_soup = Soup(req.content, "html.parser")
+            data = [film.img["alt"] for film in page_soup.find("ul", attrs={"class": "poster-list"}).findAll("li")]
+            
+            return data[:n]
+        
+        except Exception as e:
+            return e
```
Comment (nikhil25803) [inline src/scrape_up/letterboxd/letterboxd.py:76]: Return None in case of any exception

File: src/scrape_up/letterboxd/letterboxd.py [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+        try:
+            keys = ["title", "rating", "date", "review"]
+            response = get(f"{self.url}/films/reviews/by/added/", self.config)
+            soup = Soup(response.content, "html.parser")
+            reviews_list = [dict(zip(keys, [text.replace("Watched", "").strip() for text in review_item.text.strip().split("  ") if text not in ["", "\n"]])) for review_item in soup.find_all("li", attrs={"class": ["film-detail", "viewing-poster-container"]})]
+
+            return reviews_list[:n]
+        
+        except Exception as e:
+            print(f"An error occurred: {e}")
```
Comment (nikhil25803) [inline src/scrape_up/letterboxd/letterboxd.py:102]: Remove all the print statement from the code.

File: src/scrape_up/letterboxd/letterboxd.py [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+        
+        Example:
+        ```
+        letterboxd_user = Letterboxd(username="arpy8")
+        letterboxd_user.get_watchlist(n=5)
+        ```
+        
+        Returns a list of watchlisted movies by the user of length `n`.:
+        ```python
+        ['Good Time', 'Beginners', 'Y Tu Mamá También', 'Dogville', 'The House That Jack Built']
```
Comment (nikhil25803) [inline src/scrape_up/letterboxd/letterboxd.py:117]: Same here, return a dictionary with key-values pairs

File: src/scrape_up/letterboxd/letterboxd.py [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+        try:
+            req = get(self.url, self.config)
+
+            page_soup = Soup(req.content, "html.parser")
+            data = page_soup.find_all("a", attrs={"href": f"/{self.username}/following/"})[0].find("span").text
+            
+            return int(data)
+        
+        except Exception:
+            return None
```
Comment (nikhil25803) [inline src/scrape_up/letterboxd/letterboxd.py:180]: Also document your module in `dev-documentation.md` file.

**Review by nikhil25803 — CHANGES_REQUESTED** [review]
Hey @arpy8 , requested some changes, Address them and let me know.

**Comment by arpy8** [comment]
> Hey @arpy8 , requested some changes, Address them and let me know.

Hi @nikhil25803 , I've made the necessary changes, kindly take a look.
Thanks

**Review by nikhil25803 — APPROVED** [review]

**Comment by nikhil25803** [comment]
Great work @arpy8  !!
Do not forget to
+ Share your contribution on socials and tag Clueless 🫂
+ Give this repository a star ✨
+ Follow Clueless Community on GitHub ✅

**Pull request merged**